### DISCUSSION TITLE
RaspberryMatic WebUi wird nicht geladen: 502: Bad Gateway

### DISCUSSION THREAD

**Discussion by tm8** [OpenCCU/OpenCCU#1380] [🇩🇪 Fragen&Antworten] [answered]
Hallo,

da ich seit Anfang an Homematic Nutzer bin und seit vielen Jahren RaspberryMatic, möchte ich mich hier erst einmal bei Jens bedanken für die tolle Arbeit.

Ich bin zur Zeit dabei von ioBroker auf Home Assistant umzusteigen. Aus verschiedenen Gründen. Ich habe mehr über 80 Homematic und Homematic IP Komponenten aus diesem Grund habe ich jetzt versucht mit dem RaspberryMatic Addon auf Home Assistant zu testen. Ich habe nun mehr als zwei Tage jetzt versucht RaspberryMatic Addon zum laufen zu bekommen, besser gesagt es läuft, aber beim Zugriff auf die RaspberryMatic WebUI bekomme ich im Home Assistant eine "502: Bad Gateway". Ich habe viel getestet komme aber nicht weiter.

Im Log von Home Assistant wird mir bei jedem Klick auf "RaspberryMatic" in der linken Seitenleiste ein "21-08-11 16:34:10 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.33.4:8099 ssl:default [Connect call failed ('172.30.33.4', 8099)]" eingetragen.

Ich habe dreimal die VM neu installiert, bin auch beim Addon nach Anleitung vorgegangen, ohne Erfolg. Hab auch diverse andere Plugins, wie z.B. NodeRed usw. getestet, läuft alles einwandfrei.

Beim Google bin ich auch auf diesen [Link](https://github.com/jens-maus/RaspberryMatic/issues/1087) gestoßen, doch leider hilft mir das auch nicht weiter. 

Es ist bei den tests keinerlei andere RaspberryMatic oder Homematic installation im Netzwerk gelaufen.

Ich habe das USB Modul (Version 1) von Alexander Reinhart mit dem neuen Homematic FunkModul für Homematic und Homematic IP seit dieses auf dem Markt ist im Einsatz, in Verbindung mit RaspberryMatic als VM auf Proxmox, ohne Probleme. Das Modul wird lauf Log (siehe unten) auch im RaspberryMatic Addon erkannt.

Ich hoffe mir kann jemand helfen, wäre für mich sehr wichtig, ansonsten kann ich nicht von ioBroker auf Home Assistant wechseln.

![Bildschirmfoto 2021-08-12 um 07 42 27](https://user-images.githubusercontent.com/5777801/129144219-42dd3857-f6a9-46ce-bf5b-ac68598a7841.png)

Hier mal alle Informationen zu meinem System, sowie Logs von HA und RaspberryMatic Addon.

Proxmox auf einem Dell Server laufen:
![Bildschirmfoto 2021-08-11 um 17 20 05](https://user-images.githubusercontent.com/5777801/129058981-0b99da61-3bff-4b91-95cc-c60af7e9846d.png)

Home Assistant als VM laufen:
![Bildschirmfoto 2021-08-11 um 17 19 14](https://user-images.githubusercontent.com/5777801/129059143-b2732fc3-e824-4948-a0ee-f5ce67b34883.png)
![Bildschirmfoto 2021-08-11 um 17 17 44](https://user-images.githubusercontent.com/5777801/129059203-2407cdcc-2334-466a-b9c1-0ad3f8e315a5.png)
![Bildschirmfoto 2021-08-11 um 17 18 12](https://user-images.githubusercontent.com/5777801/129059241-10b6e442-a62f-4c4d-ab94-0567356ca01a.png)
![Bildschirmfoto 2021-08-11 um 17 18 47](https://user-images.githubusercontent.com/5777801/129059269-fa49efeb-3a5f-4925-9482-7aaa71d82748.png)

Home Assistant:
![Bildschirmfoto 2021-08-11 um 17 20 57](https://user-images.githubusercontent.com/5777801/129059825-9ffcd54b-f41c-4568-8157-9330de0cac77.png)
![Bildschirmfoto 2021-08-11 um 17 16 50](https://user-images.githubusercontent.com/5777801/129059890-f9667b8c-f59e-4d8b-8f10-3b3b46e54b82.png)

Log Home Assistant:
![Bildschirmfoto 2021-08-12 um 07 43 06](https://user-images.githubusercontent.com/5777801/129144171-cb33f4e2-d66d-4fd2-a329-dfcab960ba39.png)

`21-08-11 16:02:09 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
21-08-11 16:02:09 INFO (MainThread) [supervisor.resolution.fixup] Starting system autofix at state CoreState.RUNNING
21-08-11 16:02:09 INFO (MainThread) [supervisor.resolution.fixup] System autofix complete
21-08-11 16:02:18 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request initialize
21-08-11 16:02:18 INFO (MainThread) [supervisor.api.proxy] WebSocket access from a0d7b954_nodered
21-08-11 16:02:18 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request running
21-08-11 16:13:33 WARNING (SyncWorker_2) [supervisor.docker.addon] RaspberryMatic CCU running with disabled protected mode!
21-08-11 16:13:33 INFO (SyncWorker_2) [supervisor.docker.interface] Cleaning addon_e4704d1b_raspberrymatic application
21-08-11 16:13:34 INFO (SyncWorker_2) [supervisor.docker.addon] Starting Docker add-on ghcr.io/jens-maus/raspberrymatic with version 3.59.6.20210810-8090fdb
21-08-11 16:21:16 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.33.4:8099 ssl:default [Connect call failed ('172.30.33.4', 8099)]
21-08-11 16:31:38 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.33.4:8099 ssl:default [Connect call failed ('172.30.33.4', 8099)]
21-08-11 16:32:07 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
21-08-11 16:32:40 INFO (MainThread) [supervisor.snapshots] Found 2 snapshot files
21-08-11 16:32:40 INFO (MainThread) [supervisor.snapshots] Found 2 snapshot files
21-08-11 16:34:10 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.33.4:8099 ssl:default [Connect call failed ('172.30.33.4', 8099)]
21-08-11 16:34:16 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.33.4:8099 ssl:default [Connect call failed ('172.30.33.4', 8099)]
21-08-11 16:34:19 INFO (MainThread) [supervisor.api.middleware.security] /network/info access from core_ssh
21-08-11 16:34:19 INFO (MainThread) [supervisor.api.middleware.security] /network/info access from core_ssh
21-08-11 16:34:19 INFO (MainThread) [supervisor.api.middleware.security] /host/info access from core_ssh
21-08-11 16:34:19 INFO (MainThread) [supervisor.api.middleware.security] /core/info access from core_ssh
21-08-11 16:34:24 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request initialize
21-08-11 16:34:24 INFO (MainThread) [supervisor.api.proxy] WebSocket access from a0d7b954_vscode
21-08-11 16:34:24 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request running
21-08-11 17:01:13 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state CoreState.RUNNING
21-08-11 17:01:13 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.FREE_SPACE/ContextType.SYSTEM
21-08-11 17:01:13 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.PWNED/ContextType.ADDON
21-08-11 17:01:13 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.SECURITY/ContextType.CORE
21-08-11 17:01:13 INFO (MainThread) [supervisor.resolution.check] System checks complete
21-08-11 17:01:13 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.RUNNING
21-08-11 17:01:13 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
21-08-11 17:01:13 INFO (MainThread) [supervisor.resolution.fixup] Starting system autofix at state CoreState.RUNNING
21-08-11 17:01:13 INFO (MainThread) [supervisor.resolution.fixup] System autofix complete
21-08-11 17:02:07 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
21-08-11 17:16:07 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Cannot connect to host 172.30.33.4:8099 ssl:default [Connect call failed ('172.30.33.4', 8099)]
21-08-11 17:17:11 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API connection is closed
21-08-11 17:17:12 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request initialize
21-08-11 17:17:12 INFO (MainThread) [supervisor.api.proxy] WebSocket access from a0d7b954_vscode
21-08-11 17:17:12 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request running`

Log RaspberryMatic Addon:
![Bildschirmfoto 2021-08-11 um 17 16 58](https://user-images.githubusercontent.com/5777801/129143913-e396b3b9-14b2-4793-9778-9f5111e05ed6.png)
`Mounting /data as /usr/local (Home Assistant Add-On): OK
Identifying onboard hardware: oci, OK
Initializing RTC Clock: onboard, OK
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Initializing System: OK
Starting logging: OK
Init onboard LEDs: init, OK
Starting irqbalance: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.30.33.4, OK
Identifying Homematic RF-Hardware: ....HmRF: RPI-RF-MOD/HB-RF-USB@usb-0000:00:1d.0-1, HmIP: RPI-RF-MOD/HB-RF-USB@usb-0000:00:1d.0-1, OK
Updating Homematic RF-Hardware: RPI-RF-MOD: 4.2.14, OK
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting ha-proxy: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: .OK
Starting rfd: .OK
Starting HMIPServer: ........OK
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: disabled
Starting Third-Party Addons: OK
Starting crond: OK
Setup onboard LEDs: booted, OK`

configuration.yaml
![Bildschirmfoto 2021-08-12 um 07 41 31](https://user-images.githubusercontent.com/5777801/129144430-72268ed9-4f4f-4177-ba0f-c4e069c74ba0.png)

`homematic:
  interfaces:
    rf:
      host: e4704d1b-raspberrymatic
      port: 2001
      resolvenames: json
      username: Admin
      password: xxxxxxxxxx
    wired:
      host: e4704d1b-raspberrymatic
      port: 2000
      resolvenames: json
      username: Admin
      password: xxxxxxxxxx
    hmip:
      host: e4704d1b-raspberrymatic
      port: 2010
      resolvenames: json
      username: Admin
      password: xxxxxxxxxx
    groups:
      host: e4704d1b-raspberrymatic
      port: 9292
      resolvenames: json
      username: Admin
      password: xxxxxxxxxx
      path: /groups
  hosts:
    raspberrymatic:
      host: e4704d1b-raspberrymatic
      username: Admin
      password: xxxxxxxxxx`

**Comment by tm8** [comment, marked as answer]
Kann hier wirklich niemand helfen. Mir würde schon ein Tipp reichen. Danke

**Reply by jens-maus** [reply]
Dieses Problem sollte mit der nächsten RaspberryMatic beseitigt sein. Du könntest aber auch einen der nightly snapshots probieren zu nutzen - zumindest um zu kontrollieren ob das Problem in der Tat beseitigt ist.

Des Weiteren kannst du auch einfach in der RaspberryMatic Add-On Konfiguration den Port 80 z.b. auf Port 8080 umleiten lassen, dann kannst du direkt auf die RaspberryMatic WebUI indem du via Port 8080 und nicht über ingress proxy auf die WebUI von RaspberryMatic gelangst.

**Reply by tm8** [reply]
Ach ja, wie komme ich an einen "nightly snapshot"?

**Comment by tm8** [comment]
Hallo Jens, vielen Dank das werde ich testen. Ich werde hier berichten. Danke.

**Reply by jens-maus** [reply]
Na in dem du dieses Repository (https://github.com/jens-maus/RaspberryMatic) als Add-on store in HA hinzufügst natürlich. Welche url hast du den bisher verwendet? hoffentlich nicht die alte URL aus der Test/Entwicklungszeit das HA Addons. Diese URL ist nicht mehr valide und sollte für produktive Systeme nicht mehr genutzt werden!

**Reply by tm8** [reply]
Na, dann liegt da der Fehler, ich habe die URL (https://github.com/jens-maus/ha-addon-raspmatic/)verwendet. Das stand so in der Anleitung. Ok, werde mit der neuen gleich noch einmal neu installieren. Habe gerade die Portweiterleitung getestet die ginge jetzt schon.

**Reply by jens-maus** [reply]
> Na, dann liegt da der Fehler, ich habe die URL (https://github.com/jens-maus/ha-addon-raspmatic/)verwendet. Das stand so in der Anleitung. 

Welche "Anleitung" bitte? Das stand vielleicht im GitHub Ticket der ursprünglichen Entwicklungszeit so mal drin, aber in der offiziellen Anleitung (siehe https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant) unter Punkt 2 steht da eindeutig die URL dieses Repositories als die URL die man als addon repository angeben sollte.

**Reply by tm8** [reply]
Besten Dank, mit der richtigen URL funktioniert das jetzt auch. Ich hatte die Alte. Scheint zu laufen. Dann wird das heute eine lange Nacht.

Die URL stand in einem Beitrag irgendwo im Home Assistant Forum. Ich habe nach einer Möglichkeit gesucht das USB Modul von Alexander Reinert mit dem HomeMatic Addon von HA zum laufen zu bekommen.
